rewrite-maven/src/main/java/org/openrewrite/maven/internal/MavenPomDownloader.java

```diff
@@ -808,6 +758,90 @@ Collection<MavenRepository> distinctNormalizedRepositories(
         return result == null || !result.isPresent() ? null : applyAuthenticationToRepository(result.get());
     }
 
+    @Nullable
+    MavenRepository normalizeRepository(MavenRepository repository) throws Throwable {
+        // Always prefer to use https, fallback to http only if https isn't available
+        // URLs are case-sensitive after the domain name, so it can be incorrect to lowerCase() a whole URL
+        // This regex accepts any capitalization of the letters in "http"
+        String originalUrl = repository.getUri();
+        String httpsUri = originalUrl.toLowerCase().startsWith("http:") ?
+                repository.getUri().replaceFirst("[hH][tT][tT][pP]://", "https://") :
+                repository.getUri();
+        if (!httpsUri.endsWith("/")) {
+            httpsUri += "/";
+        }
+
+        HttpSender.Request.Builder request = httpSender.options(httpsUri);
+        if (repository.getTimeout() != null) {
+            request = request.withConnectTimeout(repository.getTimeout())
+                    .withReadTimeout(repository.getTimeout());
+        }
+
+        ReachabilityResult reachability = reachable(repository, applyAuthenticationToRequest(repository, request));
```

Comment:
I thought OPTIONS requests should not include any credentials, but it seems these are added here; should we remove those just to be sure?

Reply:
I dont think sending auth is a problem, but options should indeed not require it. However, you never know with custom proxy setups so I think it's OK to just leave it?